packages/tablev2/tablev2.stories.tsx

```diff
@@ -77,6 +85,82 @@ storiesOf("Table V2", module)
       )}
     </DataUpdateContainer>
   ))
+  .add("as a checkbox table", () => (
+    <DataUpdateContainer updateRateMs={1000}>
+      {({ items }) => {
+        type Action<A> =
+          | { type: "selectAll" }
+          | { type: "deselectAll" }
+          | { type: "toggle"; item: A };
+
+        function reducer(state: FakerData[], action: Action<FakerData>) {
+          switch (action.type) {
+            case "selectAll":
+              return [...items];
+            case "deselectAll":
+              return [];
+            case "toggle":
+              return state.includes(action.item)
+                ? state.filter(i => i !== action.item)
+                : state.concat([action.item]);
+          }
+        }
+
+        const [selected, dispatch] = React.useReducer(reducer, []);
+
+        // prettier-ignore
+        const headerChecked =
+          selected.length === 0            ? false :
+          selected.length === items.length ? true :
+                                           "indeterminate";
+
+        const toggle = x =>
+          React.useCallback(() => dispatch({ type: "toggle", item: x }), [x]);
+
+        return (
+          <Table
+            data={items}
+            toId={el => el.id.toString()}
+            columns={[
+              // THIS IS THE COLUMN YOU WANT TO ADD ALONG WITH THE CEREMONY ABOVE.
+              {
+                id: "checkboxCol",
+                header: (
+                  <CheckboxCell
+                    id="toggleAllRows"
+                    checked={headerChecked}
+                    inputLabel="Toggle all rows"
+                    onChange={() =>
+                      dispatch({
+                        type: headerChecked ? "deselectAll" : "selectAll"
+                      })
+                    }
+                  />
+                ),
+                // the width of a checkbox + cell-padding ( + currently 8px to not make the drag handle hide parts of the checkbox. shall we make its bg transparent?)
+                initialWidth: "44px",
```

Comment:
yeah, it would. also i totally don't like the math behind it currently, BUT your remark above resolves this as well. there's no need to specify a width, if we bake that into the first "real" column.